### PR TITLE
[No QA] Add rule to recognize user `@here` mentions

### PR DESCRIPTION
**Pull request by getusha**
@puneetlath @mananjadhav

### Fixed Issues
$ https://github.com/Expensify/App/issues/17882

# Tests
1. Open any chat
2. Send valid mention e.g. `@john`, `@david`
3. Send @here mention
The user mention should be replaced as <mention-user>mention</mention-user> and the @here should be <mention-here>@here</mention-here>

# QA

Same as tests

### CODE REVIEW

**Review by puneetlath — APPROVED** [review]
Approving since the changes are the same as in https://github.com/Expensify/expensify-common/pull/524 which I previously approved.

**Review by puneetlath — APPROVED** [review]
Approving since the changes are the same as in https://github.com/Expensify/expensify-common/pull/524 which I previously approved.

**Review by mananjadhav — APPROVED** [review]

**Comment by mananjadhav** [comment]
Verified the code changes, compared to the old PR.

**Comment by puneetlath** [comment]
Adding `No QA` since the QA will happen in the PR to incorporate the new expensify-common version in App.

**Comment by puneetlath** [comment]
PR for updating the version is here: https://github.com/Expensify/App/pull/18655

**Comment by puneetlath** [comment]
Thanks @mananjadhav would you be able to do the C+ checklist for the App PR. Since it wasn't needed for this one.

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -49,6 +49,39 @@ export default class ExpensiMark {
...
+            /**
+             * This regex matches a valid user mention in a string.
+             * A user mention is a string that starts with the '@' symbol and is followed by a valid user's primary login
+             *
+             * Note: currently we are only allowing mentions in a format of @+19728974297@expensify.sms and @username@example.com
+             * The username can contain any combination of alphanumeric letters, numbers, and underscores
+             */
+            {
+                name: 'userMentions',
+                regex: new RegExp(`((&#x60;|<code>|<pre>)\\s*?)?[\`.a-zA-Z]?@+${CONST.REG_EXP.EMAIL_PART}`, 'gm'),
```
Comment (eVoloshchak) [inline lib/ExpensiMark.js:76]: 👋 Dropping a note about this causing a regression in https://github.com/Expensify/App/issues/18857. Essentially just missed a case where mention is inside of a link. More details are [here](https://github.com/Expensify/App/issues/18857#issuecomment-1546670288)

File: lib/str.js [hunk 2]
```diff
@@ -935,6 +935,25 @@ const Str = {
...
+    /**
+     * We validate mentions by checking if it's first character is an allowed character
+     * and by checking that we make sure it isn't inside other tags where mentions aren't allowed.
+     * For example, *@username@expensify.com* is a valid mention because we allow bold styling for it,
+     * but `@username@expensify.com` is not because we do not allow mentions within code
+     *
+     * @param {String} mention
+     * @returns {bool}
+     */
+    isValidMention(mention) {
```
Comment (eVoloshchak) [inline lib/str.js:947]: Dropping a note that this has caused https://github.com/Expensify/App/issues/20085
Missed an edge case where `test+1@gmail.com@gmail.com` was rendered as mention
image
